src/broadcast/scrobblingmanager.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <QObject>
+#include <QMutex>
+#include <QLinkedList>
+
+#include "broadcast/metadatabroadcast.h"
+#include "mixer/basetrackplayer.h"
+#include "track/track.h"
+#include "track/tracktiminginfo.h"
+
+class BaseTrackPlayer;
+class PlayerManager;
+
+class ScrobblingManager : public QObject {
+    Q_OBJECT
+  public:
+    ScrobblingManager(PlayerManager *manager);
+  private:
+    struct TrackInfo {
+        TrackPointer m_pTrack;
```

Comment:
Storing shared track pointers in a class that exists for an unforeseeable time span, that is manually allocated and deleted and moreover stored as a **plain** pointer in a list managed by 2 elongate nested for loops concerns me. Remember, without releasing the track pointers you will not only introduce a memory leak, but also prevent that any modifications are written into the database or exported into files.

Reply:
Well the tracks are released as soon as they are ejected from all decks. I don't see how this is a problem since as long as there is a deck with a track the pointer is not released either.

Reply:
Are you sure that they are deleted always and promptly? The code dealing with this lifecycle management is too complex and hard to reason about. That's my concern.

Reply:
Well, I'm reasonably sure, unless there are unforeseen bugs. The problem is if I substitute it by a weak pointer then after the track has been dropped from the decks it will already be deleted, and then I can't query it further to mantain internal state.

Reply:
IMHO the weak pointer idea is not too bad.
You normally scrobble a track only if it is playing, right? So other weak pointer should always point to a valid object if you needed.
If you hit a stalled week pointer, you can debug assert, because you code is faulty if it happens.

Reply:
I can add weak pointers and then pass all the info by value to the metadata broadcaster.